### PR TITLE
Application DB, study endpoints.

### PR DESCRIPTION
**Pull request by marikomedlock**
- Setup application database, which is now used by tests and the `local-dev` scripts.
- Added a feature flag `artifact-storage-enabled` to control whether we enable artifact storage endpoints. Currently set to `false` in `application.yaml` until we have a CloudSQL deployed in the dev environment, and set to `true` for testing.
- Added endpoints for managing studies: create, update, delete, list. Studies have only 4 properties for now: `id` (uuid), `displayName`, `description`, `properties` (key-value map of strings).

### CODE REVIEW

**Review by tjennison-google — COMMENTED** [review]
Is there not a library or something that could automatically handle all of the object storage? It seems like there has to be an easier way that manually writing SQL queries.

File: api/docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -26,19 +26,32 @@ Use the key file to set the `gcloud` application default credentials.
 export GOOGLE_APPLICATION_CREDENTIALS=$(pwd)/rendered/tanagra_sa.json
 ```
 
+### Make scripts executable
+There are several helper scripts for local development. Make them all executable after pulling from GH.
+```
+chmod a+x ./api/local-dev/*
```
[the remark this answers is not in the document]
Comment (marikomedlock) [reply api/docs/CONTRIBUTING.md:32]: Done, thanks for the tip!

File: api/src/main/java/bio/terra/tanagra/app/Main.java [hunk 2]
```diff
@@ -9,8 +9,14 @@
     // We don't make use of DataSource in this application, so exclude it from scanning.
     exclude = DataSourceAutoConfiguration.class)
 @ComponentScan(
-    // Scan all packages within this service
-    basePackages = "bio.terra.tanagra")
+    basePackages = {
+      "bio.terra.tanagra",
+      "bio.terra.common.db",
+      // Logging components & configs.
+      //      "bio.terra.common.logging",
```
[the remark this answers is not in the document]
Comment (marikomedlock) [reply api/src/main/java/bio/terra/tanagra/app/Main.java:16]: Leftover from debugging. Removed.

File: api/src/main/java/bio/terra/tanagra/app/controller/StudiesV2ApiController.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            .displayName(body.getDisplayName())
+            .description(body.getDescription())
+            .properties(fromApiObject(body.getProperties()))
+            .build();
+    studyService.createStudy(studyToCreate);
+    return ResponseEntity.ok(toApiObject(studyToCreate));
+  }
+
+  @Override
+  public ResponseEntity<Void> deleteStudy(UUID studyId) {
```
[the remark this answers is not in the document]
Comment (marikomedlock) [reply api/src/main/java/bio/terra/tanagra/app/controller/StudiesV2ApiController.java:48]: Not for performance benefits, just better design to have the DB enforce requirements we expect. I'll change to strings if we move away from UUIDs for IDs.

**Comment by marikomedlock** [comment]
> Is there not a library or something that could automatically handle all of the object storage? It seems like there has to be an easier way that manually writing SQL queries.

I tried to stick as close as possible to the pattern other Terra services user (e.g. RBS, WSM). All manage the app DBs semi-manually: Liquibase handles migrations using files we write manually; Spring handles DB transactions and connections using classes we write manually.

**Review by tjennison-google — APPROVED** [review]

File: api/src/main/java/bio/terra/tanagra/app/controller/StudiesV2ApiController.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  @Autowired
+  public StudiesV2ApiController(StudyService studyService) {
+    this.studyService = studyService;
+  }
+
+  @Override
+  public ResponseEntity<ApiStudyV2> createStudy(ApiStudyCreateInfoV2 body) {
+    // Generate a random UUID for the new study ID.
+    UUID newStudyUuid = UUID.randomUUID();
```
[the remark this answers is not in the document]
Comment (marikomedlock) [reply api/src/main/java/bio/terra/tanagra/app/controller/StudiesV2ApiController.java]: Changed to a 10-character alphanumeric string. Per offline discussion, we may revisit this in the future e.g. to allow the caller to request a specific id.

**Pull request merged**